tensorflow_datasets/core/download/downloader_test.py

```diff
@@ -39,6 +39,7 @@ def __init__(self, url, content, cookies=None, headers=None, status_code=200):
     self.cookies = cookies or {}
     self.headers = headers or {'Content-length': 12345}
     self.status_code = status_code
+    self.read = self.raw.read
```

Comment:
why add this?

Reply:
@rsepassi  `_FakeResponse` is supposed to mimic both `Response` Object from `requests` and the "modified" ioBufferedReader object we get when we `urlopen` using `urllib.request`. The buffered Reader Object has `headers`, status codes, but doesn't have `iter_content` as the whole thing is read via the read function of the `BufferedReader`. So, this dummy read function, mimics the read function.

Reply:
@rse

Reply:
@rsepassi